components/suggestion/channel_mention_provider.tsx

```diff
@@ -116,20 +143,20 @@ export default class ChannelMentionProvider extends Provider {
         this.startNewRequest(prefix);
 
         const words = prefix.toLowerCase().split(/\s+/);
-        const wrappedChannelIds = {};
-        var wrappedChannels = [];
+        const wrappedChannelIds: WrappedChannelIds = {};
```

Comment:
since this is an intermittent state, lets define the type here itself
```suggestion
        const wrappedChannelIds: Record<string, boolean> = {};
```